### ISSUE TITLE
Data tables - questions about accessibility 

### ISSUE THREAD

**Issue by terribly** [wet-boew/wet-boew#7868]
Has anyone done accessibility testing on the data tables function? I have been getting some complaints about this and was trying to find out what the state of accessibility was on data-tables.net. I see that there were issues but I am not seeing that there were fixes. 

Some of the things that seem to be happening are:

There seems to be no keyboard way of activating the sort filter. 

There is no change of focus on the filter control per se but rather on the entire cell that the sort filter is in. 

Also concerned about the placement of the filter. It puts the Filter Items before the header row. I am not sure that someone using a screen reader would even know what the items are, so this would be useless at this point unless its screen reader order places it after the headings. Is this happening? I cannot tell what the screen order is.

It might be good to get some real accessibility testing with users with screenreaders and keyboard only testing to see if these are passably accessible or not, or if the sorting and filtering is causing more grief than they are worth at this point. 

If there are significant accessibility issues to using data tables, maybe there be a caveat in the Style Guide? I think we have a risk factor here for government users who are supposed to be making our tables accessible.

**Comment by kodecount** [comment]
**Sort Filter (Ascending/Descending Columns)**:

I tested with Jaws and NVDA and was able to change ascending or descending to each column. The only issue I seen is that the doesn't know the column sorting has changed. 

**Filter items**:

This was also a issue I experience on my current project. Our accessibility group wanted the results to focus. This isn't really easy to do with the built in search due to you can't really know the user is finished searching. So we went with a custom search and when we refresh the page with new content we focus on the first row in the table to show their was results found or not found.

**Keyboard Functionality**:

During out testing we found that once the table was focus the user needed to use arrow commands to go throw each column in each row. Also text wouldn't read as a paragraph if it wasn't in `<p></p>` html tags. Also we needed to add our own rules to the DataTable for it to tell the user where in the table they are located.

As for page buttons they work okay it tells them the page change, but the only downside is they have to start from the bottom of the table and go back to top then scroll through the table again to get to the page controls. If be better after a page toggle to bring focus to the first row so the user doesn't have to hear the table twice (working on this currently).

**Other**: If you wish for any code I can provide. Just not currently at work.

**Comment by kodecount** [comment]
Lot of the issue I lot discussed can't easily be fixed by WET (or are out of it's control) what could be done is some documentation to explain the best approach (like `<p></p>` or adding rules). Another option is WET doing this some how during table generation (but that could be a big task).

**Comment by Christopher-O** [comment]
I just wanted to add my voice to this issue.   Our accessibility team did mention the functionality error with filtering the content.  After hitting the "Filter" button, the focus goes to the first row and first column of the table, by-passing the datatables info about "Showing X to X of X entries".   I would recommend if possible that the script could be amended to focus on the aria-live element "polite" or even the created dataTables_info id so a screen reader can tell how many rows were filtered out.

**Comment by Christopher-O** [comment]
Just an update, I got a response from our adaptive technology team.  The solution was the aria-controls="dataset-filter" is incorrect as it bypasses the filter information. The aria-controls should be "data-set-filter_info" so that it reads the "Showing X to X of X entries".  This is optimized for JAWS 18.

**Comment by LaurentGoderre** [comment]
@SBleeks what do you think?

**Comment by terribly** [comment]
Have the issues relating to the accessibility of data tables been resolved? It does not look like it. Given this, perhaps a warning should be put in the Style Guide that these are not accessible. I am seeing more widespread use of this feature and it is affecting accessibility. 

I also noticed that the sort filters cannot be activated by a keyboard. Unless I am missing something.

**Comment by jkshapiro** [comment]
> I also noticed that the sort filters cannot be activated by a keyboard. Unless I am missing something.

@terribly I couldn't figure it out either until I checked here for existing bug reports. Turns out it's the Enter key. I agree that this needs fixing; it's not at all obvious.

**Comment by StacyBleeks** [comment]
does this issue need further clairification?
